src/__test__/app.test.ts

```diff
@@ -0,0 +1,143 @@
+import Product from '../domain/Product';
+import VendingMachine from '../domain/VendingMachine';
+import { ERROR_MESSAGE } from '../constants';
+
+const addProduct = (name: string, price: number, quantity: number) => {
+  const newProduct = {
+    name: name,
+    price: price,
+    quantity: quantity,
+  } as Product;
+
+  VendingMachine.instance.addProduct(newProduct);
+};
+
+const findProduct = (name: string): Product =>
+  VendingMachine.instance.products.find((product) => product.name === name);
+
+const updateProduct = ({ currentName, changedName, changedPrice, changedQuantity }) => {
+  VendingMachine.instance.updateProduct({
+    targetName: currentName,
+    name: changedName,
+    price: changedPrice,
+    quantity: changedQuantity,
+  });
+};
+
+const checkAlert = (errorMessage: string, callback: Function) => {
+  window.alert = jest.fn().mockReturnValue(errorMessage);
+  const spyFn = jest.spyOn(window, 'alert');
+
+  callback();
+
+  expect(spyFn).toBeCalledTimes(1);
+  expect(spyFn).toBeCalledWith(errorMessage);
+};
+
+describe('상품 관리에 성공한 경우', () => {
+  beforeEach(() => {
+    localStorage.clear();
+    VendingMachine._instance = null;
+  });
+
+  test('상품을 추가할 수 있다.', () => {
+    addProduct('잡초', 1000, 10);
+
+    expect(findProduct('잡초').name).toBe('잡초');
+  });
+
+  test('특정 상품을 수정할 수 있다.', () => {
+    addProduct('잡초', 1000, 10);
```

Comment:
```suggestion
  const weed = {
    name: '잡초',
    price: 1000,
    quantity: 10,
  };

  test('상품을 추가할 수 있다.', () => {
    addProduct(weed);

    expect(findProduct(weed.name).name).toBe(weed.name);
  });

  test('특정 상품을 수정할 수 있다.', () => {
    addProduct(weed);
```
위에 코멘트를 반영하신다면 보다 더 직관적인 테스트를 작성하실 수 있을거에요!

Reply:
오호 내부에서 객체를 만들어서 인자로 넣어주는 방법도 괜찮네요!

Reply:
반영 링크입니다! [99dcb88](https://github.com/woowacourse/javascript-vendingmachine/pull/32/commits/99dcb880b05a9ecc828ce945bd8f1b390859fe57)